app/presenters/tree_builder_alert_profile_obj.rb

```diff
@@ -0,0 +1,72 @@
+class TreeBuilderAlertProfileObj < TreeBuilder
+  def initialize(name, type, sandbox, build = true, assign = nil)
+    @assign = {
+      :new     => assign[:new],
+      :current => assign[:current]
+    }
+
+    build = false unless @assign[:new][:assign_to]
+    super(name, type, sandbox, build)
+  end
+
+  def override(node, object, _pid, _options)
+    identifier = (object.name.presence || object.description)
+    if object.kind_of?(MiddlewareServer)
+      identifier += "-" + object.hostname
+    end
+
+    if @assign[:new][:assign_to].ends_with?("-tags")
+      node[:icon] = "fa fa-tag"
+    elsif @assign[:new][:assign_to] == "tenant"
+      node[:icon] = "pficon pficon-tenant"
+    else
+      node[:image] = ActionController::Base.helpers.image_path("100/#{@assign[:new][:assign_to]}.png")
+    end
+
+    node[:title] = identifier
+    node[:cfmeNoClick] = true
+    node[:hideCheckbox] = false
```

Comment:
Override is not called for root i think, you set root by hand in set root options? And root options are called after the all overrides...